package/MDAnalysis/analysis/align.py

```diff
@@ -1368,7 +1368,10 @@ def get_atoms_byres(g, match_mask=np.logical_not(mismatch_mask)):
                         ag1.n_atoms, ag2.n_atoms)
             logger.error(errmsg)
             raise_from(SelectionError(errmsg), None)
-
+        except AttributeError:
+            errmsg=("Failed to check the mass matches")
+            warnings.warn(errmsg, category=SelectionWarning)
+            pass
         if np.any(mass_mismatches):
```

Comment:
You can't really continue with the rest of the code block after an AttributeError, because it assumes that `mass_mismatches` has been successfully created.